tools/sof_perf_analyzer.py

```diff
@@ -204,8 +205,29 @@ def process_kmsg_file():
                 next_line = next(f)
                 widget_id = next_line.split('|')[0].split(':')[-1].strip()
                 # convert to the same ID format in mtrace
-                widget_id = int('0x' + widget_id[-6:], 16)
+                try:
+                    widget_id = int('0x' + widget_id[-6:], 16)
+                except ValueError:
+                    print("no valid widget ID found, try with bind information")
+                    not_found[widget_name] = ppln_id
+                    continue
                 comp_name[str(Component(ppln_id, widget_id))] = widget_name
+            elif match_obj := re.search(r' bind ', line):
+                if not_found:
+                    # parse current line to get widget name
+                    span_end_pos = match_obj.span()[1] - 1
+                    line_split = line[span_end_pos + 1:].split()
+                    widget_name = line_split[0].split(':')[0]
+                    # parse next line get widget ID
+                    next_line = next(f)
+                    widget_id = next_line.split('|')[0].split(':')[-1].strip()
+                    widget_id = int('0x' + widget_id[-6:], 16)
+                    # get ppln_id from dictionary
+                    if not_found[widget_name]:
+                        ppln_id = not_found[widget_name]
```

Comment:
`if ppln_id := not_found[widget_name]`, this removes the repetition of `not_found[widget_name]`

Reply:
at this moment, ppln_id is still valid? 
if not_found[widget_name]
this condition means: with current parsed widget_name, if meet this condition, means we need setup dictionary for this widget, any logic wrong?